### ISSUE TITLE
Issue installing. Directory missing

### ISSUE THREAD

**Issue by jolubevi** [Myddleware/myddleware#1049]
Hi, I've installed following the next steps:

git clone https://github.com/Myddleware/myddleware.git
cd myddleware
composer install 
yarn install 
yarn build
After do this, I try to access MYIP/myddleware/public and after press "Install" or "Login" the web redirects to IP/myddleware/public/install/requirements and got the error "Not Found"

**Comment by CamillePMyddleware** [comment]
Hello @jolubevi, 
can you tell me which version of PHP you are using? If you want to install the latest version of Myddleware, you need PHP 8 or higher. You can also check the requirements by running this command in your terminal: "symfony check:requirements". Could you also send us more information, such as a screenshot, console error, or log? Alternatively, you can try to install Myddleware using your terminal.

Feel free to reach out if you need additional assistance.
Thanks, and have a nice day

**Comment by jolubevi** [comment]
Hi,

I add some data and screenshots:

PHP 8.1.21 (cli) (built: Jul  8 2023 07:10:19) (NTS)

Simfony check
![image](https://github.com/Myddleware/myddleware/assets/58666166/5d4a8209-fd37-4e59-85a9-cd8130b88dc7)

html/myddleware/public directory just contains:

![image](https://github.com/Myddleware/myddleware/assets/58666166/82ba4a82-16dd-48ba-ac84-e1c28ac04642)

When I try IP/myddleware/public I get:
![image](https://github.com/Myddleware/myddleware/assets/58666166/7996d75f-8fe3-4cfb-822e-e8c60d076c2c)

But went I try sign in or press "Install Myddleware"...
![image](https://github.com/Myddleware/myddleware/assets/58666166/203a8e2a-abdc-411a-8048-0d45d21ba33e)

Thanks for your time!! I really appreciate that

**Comment by AlexMyddleware** [comment]
hello @jolubevi, could you provide more information about your setup ? 
What do you have in your prod.log ?
can you look at you node.js version ?
Could you try to run this command and report to us if it changes anything ?

rsync -Rr ./build/ ./build/

It will create a copy of the build folder into the build folder. It might solve the problem. At any rate, keep us posted and provide as much information as you can.

**Comment by jolubevi** [comment]
Hi Alex,

Thanks for your repply!

Answering your questions:

- I haven't found any prod.log in my system

Nodejs:
![image](https://github.com/Myddleware/myddleware/assets/58666166/2cb6d0d8-96f2-4c56-88a8-9f9264b43570)

Running the command that you suggest I got an error:

![image](https://github.com/Myddleware/myddleware/assets/58666166/e494e65b-4d94-4f17-bcfa-340257cef803)

Thanks a lot!

**Comment by AlexMyddleware** [comment]
okay @jolubevi, no issue with your node version. 

Also, you might try installing by following the terminal install section. It might work better for you. 

I would suggest deleting your whole myddleware directory, restart from scratch with the git clone, composer install, yarn install yarn build and then you can start the terminal section

![image](https://github.com/Myddleware/myddleware/assets/106162060/18d3459f-b7cc-4a70-88eb-ccab9b7df3dc)

Before starting the terminal section, after the cloning (before or after the comp/yarn section it doesn't matter) I would suggest creating an empty database then setting the .env to the proper database name and your connection too. Remember to do that too in your .env.local.

Keep me posted !

**Comment by jolubevi** [comment]
Ok, I'll try that and I'll let you know

**Comment by jolubevi** [comment]
Here are the steps (I already have the database and the user created)

Git:
![image](https://github.com/Myddleware/myddleware/assets/58666166/68d28b98-dfbc-498c-93dd-2baac85e6d79)

cd myddleware

Composer install:
![image](https://github.com/Myddleware/myddleware/assets/58666166/b21c84f5-c6e1-4381-a8db-242a31a4ae2f)

Yarn install:
![image](https://github.com/Myddleware/myddleware/assets/58666166/aff6c69c-0da1-4ab5-bfe7-74e44291a8ff)

Yarn build:
![image](https://github.com/Myddleware/myddleware/assets/58666166/62318721-0393-4b2d-b316-64a92f14cd6f)

Editing .env file:
php bin/console doctrine:schema:update --force
php bin/console doctrine:fixtures:load --append
php bin/console myddleware:add-user

But when I try Sign in...

![image](https://github.com/Myddleware/myddleware/assets/58666166/3fa846af-2d56-405e-a514-857d652050d7)

Thanks again for your support. I don't know what I'm doing wrong

**Comment by AlexMyddleware** [comment]
hello @jolubevi well this is very strange indeed. 

- There might something wrong with url, in the sense that given your current environment, xxxxxxxx/myddleware/public/rule/panel might not be exactly the right url you need to use to get to the app, but I can't advise you since i don't know your folder structure.

 - However that is unlikely. If you manage again to go to the login page, then it's probably not that. This leaves us with what's the most likely problem: file permissions. I would not have that problem since i'm on windows with laragon and all of that is done automatically, but on your specific configuration maybe something prevents the server to access the specific php file for instance, which results in a not found error. Please try to grant higher access to your folder and subfolders and files recursively. 
 
 - Do you have any logs ? dev.log or prod.log ? check that and if you do please post the latest messages

**Comment by Myddleware** [comment]
Thanks @AlexMyddleware ,
Yes you are right, @jolubevi  could you make sure your webuser (www-data if you use apache) has the permission to write in the folders var/logs and var/cache

**Comment by JiahuiLuData** [comment]
Hi support team, I am also having this exact problem(even for the missing logo on login page), and I have followed every step above and installed/uninstalled it many times. I have set the permission to write in the folders var/logs and var/cache, but the page is still not found. Do you know what to do after that?

 Neither dev.log or prod.log updated during the time the error happened. 

In case this is a folder structure issue, I have attached my folder structure:
├── assets
├── bin
├── composer.json
├── composer.lock
├── config
├── docker
├── docker-compose.yml
├── Dockerfile
├── docs
├── fixtures
├── LICENSE
├── Makefile
├── migrations
├── node_modules
├── package.json
├── phpunit.xml.dist
├── phpunit.xml.dist.bak
├── public
├── README.md
├── src
├── symfony.lock
├── templates
├── tests
├── translations
├── var
├── vendor
├── webpack.config.js
└── yarn.lock
and within folder public, here is what it looks like: 
├── public
│   ├── build
│   ├── favicon.ico
│   ├── index.php
│   ├── js
│   └── robots.txt
Could you please advise the next step? Very appreciate your help!
@Myddleware @AlexMyddleware @RomanFaure @CamillePMyddleware

**Comment by Myddleware** [comment]
Hi @JiahuiLuData , 
Thanks for you interest in Myddleware.
@AlexMyddleware or @CamillePMyddleware will help you on Monday. 
Have a nice weekend.

**Comment by JiahuiLuData** [comment]
Sure thanks! Just to let you know we even copied the entire directory structure from our working production environment and that doesn't work either.

**Comment by danielavetica** [comment]
i've got the exact same issue...  
Run the setup step-by step by the instruction, get the inlogpage:
![image](https://github.com/Myddleware/myddleware/assets/9462501/370cb6f7-9474-4e7a-a2cb-cc707f47eb89)

but when I click install:
![image](https://github.com/Myddleware/myddleware/assets/9462501/ce9aac7b-3968-4a92-a063-5520ab95c589)

**Comment by JiahuiLuData** [comment]
Hi @Myddleware, I am still waiting for your help!
@Myddleware @AlexMyddleware @RomanFaure @CamillePMyddleware

**Comment by AlexMyddleware** [comment]
Hi @JiahuiLuData , @danielavetica, Let's get into it. What does your .env and .env.local look like ? Whether you have one or not is not the problem, but if you do, I would like to check the content (don't put the login/password or secret of course, put fake values.

Do you have any kind of logs ? I would assume your environment is prod so see if you have any prod.log somewhere.

Can you guys provide a detailed ls ? The best would be ls -alps of the root folder, src and public. Here are mine for comparison.
root:
$ ls -alps
total 1329
 12 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 09:27 ./
 16 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:27 ../
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096     92 Aug 31 13:38 .dockerignore
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   2352 Sep 11 09:27 .env
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    279 Sep  6 15:37 .env.local
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    221 Aug 31 13:38 .env.test
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096     14 Aug 31 13:38 .ftpquota
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 16:46 .git/
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   2390 Aug 31 13:38 .gitignore
 16 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096  14575 Aug 31 13:38 .php-cs-fixer.cache
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    292 Aug 31 13:38 .php-cs-fixer.dist.php
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    296 Aug 31 13:38 .php-cs-fixer.php
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096      6 Aug 31 13:38 .php-version
  0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:54 .vscode/
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096     26 Aug 31 13:38 .yarnrc.yml
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 09:27 assets/
  0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 bin/
  8 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   5433 Aug 31 13:38 composer.json
552 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096 564842 Aug 31 13:38 composer.lock
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep  6 15:37 config/
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 docker/
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    148 Aug 31 13:38 docker-compose.override.yml
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   2305 Aug 31 13:38 docker-compose.yml
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   1938 Aug 31 13:38 Dockerfile
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 docs/
  0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 fixtures/
 36 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096  35815 Aug 31 13:38 LICENSE
  8 -rwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096   6195 Aug 31 13:38 Makefile
  0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 migrations/
256 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:40 node_modules/
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   1368 Sep 11 09:27 package.json
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   1397 Aug 31 13:38 phpunit.xml.dist
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    920 Aug 31 13:38 phpunit.xml.dist.bak
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:41 public/
 12 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096  10407 Aug 31 13:38 README.md
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 src/
 24 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096  20542 Aug 31 13:38 symfony.lock
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 09:27 templates/
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:38 tests/
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 09:27 translations/
  0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:41 var/
 16 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:39 vendor/
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   2992 Aug 31 13:38 webpack.config.js
292 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096 298492 Sep 11 09:27 yarn.lock

src:

$ ls -alps
total 114
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 ./
12 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 ../
 1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    13 Aug 31 13:38 .htaccess
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 Command/
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 Controller/
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 DataFixtures/
 8 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 Entity/
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 EventListener/
 0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 EventSubscribers/
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 Form/
 4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096  2137 Aug 31 13:38 Kernel.php
36 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096 35821 Aug 31 13:38 licence.txt
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 Manager/
 1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    40 Aug 31 13:38 readme.txt
 8 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 Repository/
 0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 Security/
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 Service/
 8 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep 11 09:27 Solutions/
 4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Sep  6 15:25 Templates/
 0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 Twig/
 0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096     0 Aug 31 13:38 Utils/
 
public:
 $ ls -alps
total 209
  4 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:41 ./
 12 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 09:27 ../
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096   1059 Aug 31 13:38 .htaccess
  8 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Sep 11 14:11 build/
176 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096 177470 Aug 31 13:38 favicon.ico
  4 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    832 Aug 31 13:38 index.php
  0 drwxr-xr-x 1 AzureAD+AlexisTroïtzky 4096      0 Aug 31 13:41 js/
  1 -rw-r--r-- 1 AzureAD+AlexisTroïtzky 4096    110 Aug 31 13:38 robots.txt
  
  let us compare the files and file rights for all of this for both of you, that might give us some insight to begin with

**Comment by danielavetica** [comment]
hi Alex,

this is my .env file (I had set it to local & debug=true to get some detailed information during install)9:

```
MYDDLEWARE_VERSION=3.3.0b

APP_SECRET=SomeSecretThat1Plac3dHere!
APP_ENV=local
APP_DEBUG=true
# MAILER_URL=null://

###> Delivery Address ###
DELIVERY_ADDRESS=beheer@avetica.nl
###< Delivery Address ###

COMPOSER_MEMORY_LIMIT=-1

#  * .env                contains default values for the environment variables needed by the app
#  * .env.local          uncommitted file with local overrides
#  * .env.$APP_ENV       committed environment-specific defaults
#  * .env.$APP_ENV.local uncommitted environment-specific overrides
#
# Real environment variables win over .env files.
#
# DO NOT DEFINE PRODUCTION SECRETS IN THIS FILE NOR IN ANY OTHER COMMITTED FILES.
# https://symfony.com/doc/current/configuration/secrets.html
#
# Run "composer dump-env prod" to compile .env files for production use (requires symfony/flex >=1.2).
# https://symfony.com/doc/current/best_practices.html#use-environment-variables-for-infrastructure-configuration

###> doctrine/doctrine-bundle ###
# Format described at https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/configuration.html#connecting-using-a-url
# IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
#
# DATABASE_URL="sqlite:///%kernel.project_dir%/var/data.db"
#DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7"
DATABASE_URL="mysql://myddleware:MyDBPASSWORD@localhost:3306/myddleware"
#DATABASE_URL="postgresql://app:!ChangeMe!@127.0.0.1:5432/app?serverVersion=14&charset=utf8"
###< doctrine/doctrine-bundle ###

###> nelmio/cors-bundle ###
CORS_ALLOW_ORIGIN='^https?://(localhost|127\.0\.0\.1)(:[0-9]+)?$'
###< nelmio/cors-bundle ###

###> lexik/jwt-authentication-bundle ###
JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
JWT_PASSPHRASE=SOMEJWTPASSPHRASE
###< lexik/jwt-authentication-bundle ###



###> symfony/swiftmailer-bundle ###
# For Gmail as a transport, use: "gmail://username:password@localhost"
# For a generic SMTP server, use: "smtp://localhost:25?encryption=&auth_mode="
# Delivery is disabled by default via "null://localhost"
MAILER_URL=null://localhost
MAILER_FROM=no-reply@myddleware.com
# MAILER_URL=smtp://localhost:465?encryption=ssl&auth_mode=login&username=root&password=
###< symfony/swiftmailer-bundle ###
```


The LS of my Public_html folder:
```

-rw-r--r--   1 root root   92 Jul 19  2022 .dockerignore
-rw-r--r--   1 root root 2.4K Sep 11 14:47 .env
-rw-r--r--   1 root root  221 Jul 19  2022 .env.test
-rw-r--r--   1 root root   14 Jul 19  2022 .ftpquota
drwxr-xr-x   7 root root 4.0K Jan 10  2023 .git
-rw-r--r--   1 root root 2.4K Jan  9  2023 .gitignore
-rw-r--r--   1 root root  15K Nov 21  2022 .php-cs-fixer.cache
-rw-r--r--   1 root root  292 Jul 19  2022 .php-cs-fixer.dist.php
-rw-r--r--   1 root root  296 Nov  8  2022 .php-cs-fixer.php
-rw-r--r--   1 root root    6 Nov  8  2022 .php-version
-rw-r--r--   1 root root   26 Jul 19  2022 .yarnrc.yml
-rw-r--r--   1 root root 5.8K Nov  8  2022 Dockerfile
-rw-r--r--   1 root root  35K Jul 19  2022 LICENSE
-rw-r--r--   1 root root 6.1K Jul 19  2022 Makefile
-rw-r--r--   1 root root  11K Nov  8  2022 README.md
drwxr-xr-x   8 root root 4.0K Jan 10  2023 assets
drwxr-xr-x   2 root root 4.0K Jan 10  2023 bin
-rw-r--r--   1 root root 5.2K Sep 11 14:59 composer.json
-rw-r--r--   1 root root 528K Sep 11 14:28 composer.lock
drwxr-xr-x   5 root root 4.0K Jan 10  2023 config
drwxr-xr-x   9 root root 4.0K Jan 10  2023 docker
-rw-r--r--   1 root root  148 Jan  9  2023 docker-compose.override.yml
-rw-r--r--   1 root root 2.3K Jul 19  2022 docker-compose.yml
drwxr-xr-x   4 root root 4.0K Jan 10  2023 docs
drwxr-xr-x   2 root root 4.0K Jan 10  2023 fixtures
drwxr-xr-x   2 root root 4.0K Jan 10  2023 migrations
drwxr-xr-x 589 root root  20K Sep 11 14:29 node_modules
-rw-r--r--   1 root root 1.4K Sep 11 14:59 package.json
-rw-r--r--   1 root root 1.4K Jul 19  2022 phpunit.xml.dist
-rw-r--r--   1 root root  920 Jul 19  2022 phpunit.xml.dist.bak
drwxr-xr-x   5 root root 4.0K Jan 10  2023 public
drwxr-xr-x  17 root root 4.0K Jan 10  2023 src
-rw-r--r--   1 root root  20K Jan  9  2023 symfony.lock
drwxr-xr-x  18 root root 4.0K Jan 10  2023 templates
drwxr-xr-x   3 root root 4.0K Jan 10  2023 tests
drwxr-xr-x   3 root root 4.0K Jan 10  2023 translations
drwxrwxrwx   5 root root 4.0K Sep 11 14:29 var
drwxr-xr-x  62 root root 4.0K Sep 11 14:28 vendor
-rw-r--r--   1 root root 2.9K Nov  8  2022 webpack.config.js
-rw-r--r--   1 root root 294K Jan  9  2023 yarn.lock
```


the LS of SRC:
```
-rw-r--r--  1 root root   13 Jul 19  2022 .htaccess
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Command
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Controller
drwxr-xr-x  2 root root 4.0K Jan 10  2023 DataFixtures
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Entity
drwxr-xr-x  2 root root 4.0K Jan 10  2023 EventListener
drwxr-xr-x  2 root root 4.0K Jan 10  2023 EventSubscribers
drwxr-xr-x  5 root root 4.0K Jan 10  2023 Form
-rw-r--r--  1 root root 2.1K Jul 19  2022 Kernel.php
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Manager
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Repository
drwxr-xr-x  3 root root 4.0K Jan 10  2023 Security
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Service
drwxr-xr-x  3 root root 4.0K Jan 10  2023 Solutions
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Templates
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Twig
drwxr-xr-x  2 root root 4.0K Jan 10  2023 Utils
-rw-r--r--  1 root root  35K Jul 19  2022 licence.txt
-rw-r--r--  1 root root   40 Jul 19  2022 readme.txt
```

the LS of PUBLIC:

```
-rw-r--r--  1 root root 1.1K Jul 19  2022 .htaccess
drwxr-xr-x  3 root root 4.0K Sep 11 15:00 build
drwxr-xr-x  5 root root 4.0K Jan 10  2023 bundles
-rw-r--r--  1 root root 174K Dec 15  2022 favicon.ico
-rw-r--r--  1 root root  832 Jul 20  2022 index.php
drwxr-xr-x  2 root root 4.0K Jan 10  2023 js
-rw-r--r--  1 root root  110 Jul 19  2022 robots.txt

```

**Comment by JiahuiLuData** [comment]
My .env file:

```
APP_SECRET=Thissecretisnotsosecretchangeit
APP_ENV=prod
APP_DEBUG=false
# MAILER_URL=null://

###> Delivery Address ###
DELIVERY_ADDRESS=example@email.com
###< Delivery Address ###

COMPOSER_MEMORY_LIMIT=-1

#  * .env                contains default values for the environment variables needed by the app
#  * .env.local          uncommitted file with local overrides
#  * .env.$APP_ENV       committed environment-specific defaults
#  * .env.$APP_ENV.local uncommitted environment-specific overrides
#
# Real environment variables win over .env files.
#
# DO NOT DEFINE PRODUCTION SECRETS IN THIS FILE NOR IN ANY OTHER COMMITTED FILES.
# https://symfony.com/doc/current/configuration/secrets.html
#
# Run "composer dump-env prod" to compile .env files for production use (requires symfony/flex >=1.2).
# https://symfony.com/doc/current/best_practices.html#use-environment-variables-for-infrastructure-configuration

###> doctrine/doctrine-bundle ###
# Format described at https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/configuration.html#connecting-using-a-url
# IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
#
# DATABASE_URL="sqlite:///%kernel.project_dir%/var/data.db"
DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7"
#DATABASE_URL="postgresql://app:!ChangeMe!@127.0.0.1:5432/app?serverVersion=14&charset=utf8"
###< doctrine/doctrine-bundle ###

###> nelmio/cors-bundle ###
CORS_ALLOW_ORIGIN='^https?://(localhost|127\.0\.0\.1)(:[0-9]+)?$'
###< nelmio/cors-bundle ###

###> lexik/jwt-authentication-bundle ###
JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
JWT_PASSPHRASE=PASSPHRASE
###< lexik/jwt-authentication-bundle ###



###> symfony/swiftmailer-bundle ###
# For Gmail as a transport, use: "gmail://username:password@localhost"
# For a generic SMTP server, use: "smtp://localhost:25?encryption=&auth_mode="
# Delivery is disabled by default via "null://localhost"
MAILER_URL=null://localhost
MAILER_FROM=no-reply@myddleware.com
# MAILER_URL=smtp://localhost:465?encryption=ssl&auth_mode=login&username=root&password=
###< symfony/swiftmailer-bundle ###

```

.env.local:
```
DATABASE_URL="mysql://myddleware:MYPASSWORD@localhost:3306/myddleware"
APP_ENV=prod
APP_DEBUG=false
APP_SECRET=MYSECRET
```



Root ls:

```
techsupport@localhost:/var/www/html/myddleware$ ls -al
total 1084
drwxr-xr-x  18 www-data www-data   4096 Sep  8 15:22 .
drwxr-xr-x   7 www-data www-data   4096 Sep  8 15:02 ..
drwxr-xr-x   8 www-data www-data   4096 Sep  8 15:02 assets
drwxr-xr-x   2 www-data www-data   4096 Sep  8 15:02 bin
-rw-r--r--   1 www-data www-data   5433 Sep  8 15:02 composer.json
-rw-r--r--   1 www-data www-data 550221 Sep  8 15:02 composer.lock
drwxr-xr-x   5 www-data www-data   4096 Sep  8 15:02 config
drwxr-xr-x   9 www-data www-data   4096 Sep  8 15:02 docker
-rw-r--r--   1 www-data www-data    140 Sep  8 15:02 docker-compose.override.yml
-rwxr-xr-x   1 www-data www-data   2218 Sep  8 15:02 docker-compose.yml
-rw-r--r--   1 www-data www-data   1896 Sep  8 15:02 Dockerfile
-rw-r--r--   1 www-data www-data     86 Sep  8 15:02 .dockerignore
drwxr-xr-x   4 www-data www-data   4096 Sep  8 15:02 docs
-rw-r--r--   1 www-data www-data   2298 Sep  8 15:02 .env
-rw-r--r--   1 www-data www-data    158 Sep  8 15:22 .env.local
-rw-r--r--   1 www-data www-data    215 Sep  8 15:02 .env.test
drwxr-xr-x   2 www-data www-data   4096 Sep  8 15:02 fixtures
-rw-r--r--   1 www-data www-data     13 Sep  8 15:02 .ftpquota
drwxr-xr-x   8 www-data www-data   4096 Sep  8 15:02 .git
-rw-r--r--   1 www-data www-data   2259 Sep  8 15:02 .gitignore
-rw-r--r--   1 www-data www-data  35141 Sep  8 15:02 LICENSE
-rw-r--r--   1 www-data www-data   6013 Sep  8 15:02 Makefile
drwxr-xr-x   2 www-data www-data   4096 Sep  8 15:02 migrations
drwxr-xr-x 589 www-data www-data  20480 Sep  8 15:04 node_modules
-rw-r--r--   1 www-data www-data   1330 Sep  8 15:02 package.json
-rw-r--r--   1 www-data www-data  14575 Sep  8 15:02 .php-cs-fixer.cache
-rw-r--r--   1 www-data www-data    278 Sep  8 15:02 .php-cs-fixer.dist.php
-rw-r--r--   1 www-data www-data    282 Sep  8 15:02 .php-cs-fixer.php
-rw-r--r--   1 www-data www-data   1355 Sep  8 15:02 phpunit.xml.dist
-rw-r--r--   1 www-data www-data    889 Sep  8 15:02 phpunit.xml.dist.bak
-rw-r--r--   1 www-data www-data      6 Sep  8 15:02 .php-version
drwxr-xr-x   4 www-data www-data   4096 Sep  8 15:14 public
-rw-r--r--   1 www-data www-data  10188 Sep  8 15:02 README.md
drwxr-xr-x  17 www-data www-data   4096 Sep  8 15:02 src
-rwxr-xr-x   1 www-data www-data  19777 Sep  8 15:02 symfony.lock
drwxr-xr-x  18 www-data www-data   4096 Sep  8 15:02 templates
drwxr-xr-x   3 www-data www-data   4096 Sep  8 15:02 tests
drwxr-xr-x   3 www-data www-data   4096 Sep  8 15:02 translations
drwxr-xr-x   5 www-data www-data   4096 Sep  8 15:14 var
drwxr-xr-x  63 www-data www-data   4096 Sep  8 15:04 vendor
-rwxr-xr-x   1 www-data www-data   2907 Sep  8 15:02 webpack.config.js
-rw-r--r--   1 www-data www-data 294548 Sep  8 15:02 yarn.lock
-rw-r--r--   1 www-data www-data     25 Sep  8 15:02 .yarnrc.yml
```
public ls:

```
techsupport@localhost:/var/www/html/myddleware$ ls -al public/
total 204
drwxr-xr-x  4 www-data www-data   4096 Sep  8 15:14 .
drwxr-xr-x 18 www-data www-data   4096 Sep  8 15:22 ..
drwxr-xr-x  3 www-data www-data   4096 Sep  8 15:15 build
-rw-r--r--  1 www-data www-data 177470 Sep  8 15:02 favicon.ico
-rwxr-xr-x  1 www-data www-data   1025 Sep  8 15:02 .htaccess
-rw-r--r--  1 www-data www-data    805 Sep  8 15:02 index.php
drwxr-xr-x  2 www-data www-data   4096 Sep  8 15:14 js
-rw-r--r--  1 www-data www-data    106 Sep  8 15:02 robots.txt
```
src ls:
```
techsupport@localhost:/var/www/html/myddleware$ ls -al src
total 116
drwxr-xr-x 17 www-data www-data  4096 Sep  8 15:02 .
drwxr-xr-x 18 www-data www-data  4096 Sep  8 15:22 ..
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Command
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Controller
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 DataFixtures
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Entity
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 EventListener
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 EventSubscribers
drwxr-xr-x  6 www-data www-data  4096 Sep  8 15:02 Form
-rw-r--r--  1 www-data www-data    13 Sep  8 15:02 .htaccess
-rw-r--r--  1 www-data www-data  2083 Sep  8 15:02 Kernel.php
-rw-r--r--  1 www-data www-data 35147 Sep  8 15:02 licence.txt
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Manager
-rw-r--r--  1 www-data www-data    39 Sep  8 15:02 readme.txt
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Repository
drwxr-xr-x  3 www-data www-data  4096 Sep  8 15:02 Security
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Service
drwxr-xr-x  3 www-data www-data  4096 Sep  8 15:02 Solutions
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Templates
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Twig
drwxr-xr-x  2 www-data www-data  4096 Sep  8 15:02 Utils
```

Prod.log(I was creating a duplicated user so I guess this is not very relevant to the error I am getting on webpage):

```
[2023-09-08T15:26:48.881494+00:00] console.CRITICAL: Error thrown while running command "myddleware:add-user". Message: "There is already a user registered with the "myddleware_admin" username." {"exception":"[object] (Symfony\\Component\\Console\\Exception\\RuntimeException(code: 0): There is already a user registered with the \"myddleware_admin\" username. at /var/www/html/myddleware/src/Command/AddUserCommand.php:215)","command":"myddleware:add-user","message":"There is already a user registered with the \"myddleware_admin\" username."} []
[2023-09-08T15:26:48.881494+00:00] console.CRITICAL: Error thrown while running command "myddleware:add-user". Message: "There is already a user registered with the "myddleware_admin" username." {"exception":"[object] (Symfony\\Component\\Console\\Exception\\RuntimeException(code: 0): There is already a user registered with the \"myddleware_admin\" username. at /var/www/html/myddleware/src/Command/AddUserCommand.php:215)","command":"myddleware:add-user","message":"There is already a user registered with the \"myddleware_admin\" username."} []
[2023-09-08T15:26:48.913311+00:00] console.DEBUG: Command "myddleware:add-user" exited with code "1" {"command":"myddleware:add-user","code":1} []
~                                                                                                              
```                                      
Error log from apache:
sudo less /var/log/apache2/error.log
```
[Tue Sep 12 13:57:41.414085 2023] [core:info] [pid 40666] [client 209.150.51.216:49782] AH00128: File does not exist: /var/www/html/myddleware/public/build/build/images/logo.24a56e6e.png, referer: https://MYSERVER/myddleware/public/build/login.bf3d28a3.css
[Tue Sep 12 13:57:41.422024 2023] [core:info] [pid 40668] [client 209.150.51.216:49784] AH00128: File does not exist: /var/www/html/myddleware/public/build/build/images/logo_empty.0877ec01.png, referer: https://MYSERVER/myddleware/public/build/login.bf3d28a3.css
[Tue Sep 12 13:57:53.440483 2023] [core:info] [pid 46327] [client 209.150.51.216:49790] AH00128: File does not exist: /var/www/html/myddleware/public/build/build/images/logo.24a56e6e.png, referer: https://MYSERVER/myddleware/public/build/login.bf3d28a3.css
[Tue Sep 12 13:57:53.443332 2023] [core:info] [pid 46326] [client 209.150.51.216:49789] AH00128: File does not exist: /var/www/html/myddleware/public/build/build/images/logo_empty.0877ec01.png, referer: https://MYSERVER/myddleware/public/build/login.bf3d28a3.css
[Tue Sep 12 13:58:16.025001 2023] [core:info] [pid 40664] [client 209.150.51.216:49799] AH00128: File does not exist: /var/www/html/myddleware/public/rule/panel, referer: https://MYSERVER/myddleware/public/
[Tue Sep 12 13:58:35.282994 2023] [reqtimeout:info] [pid 40667] [client 209.150.51.216:49800] AH01382: Request header read timeout

```
Seems like there missed something in build folder in error.log. So I ls the build folder as well:
techsupport@localhost:/var/www/html/myddleware/public/build$ ls -al
```
total 2420
drwxr-xr-x 3 www-data www-data    4096 Sep  8 15:15 .
drwxr-xr-x 4 www-data www-data    4096 Sep  8 15:14 ..
-rw-r--r-- 1 www-data www-data   13789 Sep  8 15:15 330.6f11cb50.js
-rw-r--r-- 1 www-data www-data  122960 Sep  8 15:15 37.9a402191.js
-rw-r--r-- 1 www-data www-data      71 Sep  8 15:15 37.9a402191.js.LICENSE.txt
-rw-r--r-- 1 www-data www-data   16328 Sep  8 15:15 37.cd9a310f.css
-rw-r--r-- 1 www-data www-data  160326 Sep  8 15:15 405.1c0254a2.js
-rw-r--r-- 1 www-data www-data  193152 Sep  8 15:15 405.93f2e0f0.css
-rw-r--r-- 1 www-data www-data    6347 Sep  8 15:15 50.7d6ccf60.js
-rw-r--r-- 1 www-data www-data  102282 Sep  8 15:15 674.0ed36a4f.js
-rw-r--r-- 1 www-data www-data   34051 Sep  8 15:15 674.24985629.css
-rw-r--r-- 1 www-data www-data 1454360 Sep  8 15:15 710.dcd41238.js
-rw-r--r-- 1 www-data www-data    4486 Sep  8 15:15 710.dcd41238.js.LICENSE.txt
-rw-r--r-- 1 www-data www-data   90005 Sep  8 15:15 755.5a8586e9.js
-rw-r--r-- 1 www-data www-data     475 Sep  8 15:15 755.5a8586e9.js.LICENSE.txt
-rw-r--r-- 1 www-data www-data     155 Sep  8 15:15 app.9e977079.js
-rw-r--r-- 1 www-data www-data    1960 Sep  8 15:15 connector.93c26766.js
-rw-r--r-- 1 www-data www-data    2207 Sep  8 15:15 entrypoints.json
-rw-r--r-- 1 www-data www-data    8709 Sep  8 15:15 fiche.e94e4977.js
-rw-r--r-- 1 www-data www-data    8633 Sep  8 15:15 filter.45e7c564.js
drwxr-xr-x 6 www-data www-data    4096 Sep  8 15:15 images
-rw-r--r-- 1 www-data www-data     650 Sep  8 15:15 install.9c1dd012.css
-rw-r--r-- 1 www-data www-data     217 Sep  8 15:15 install.d37421c8.js
-rw-r--r-- 1 www-data www-data  160818 Sep  8 15:15 login.bf3d28a3.css
-rw-r--r-- 1 www-data www-data    1080 Sep  8 15:15 login.e66c4b1c.js
-rw-r--r-- 1 www-data www-data   10684 Sep  8 15:15 manifest.json
-rw-r--r-- 1 www-data www-data     336 Sep  8 15:15 rulelist.e0ca4ec1.js
-rw-r--r-- 1 www-data www-data    1420 Sep  8 15:15 runtime.e20bb3c7.js

```
Thank you for your time and hope you find that useful for troubleshooting.

**Comment by AlexMyddleware** [comment]
@danielavetica, your APP_ENV=local is wrong, can you set it to either prod or dev ? Also i don't think the delivery address could give any issues but can you disable it just in case ?

aside from that, I don't see anything specific that's bad about your respective setups, so we are going to have to dig deeper. 

Could you @JiahuiLuData and @danielavetica try and clone our 3.3.4 version ? There is a chance that some bug in 3.3 is preventing you guys from installing, and it's been fixed in 334

**Comment by JiahuiLuData** [comment]
Hi Alex ,
Thanks for your reply. I have cloned the latest version and compared one by one on the file permissions you provided. It didn't help. Would you mind giving me some further advice?

Thanks

**Comment by danielavetica** [comment]
same here. No luck installing, still the "404 not found" message on installing.

--> just a quick (other) question: Will Myddleware be attending the global MoodleMoot in Barcelona next week?

**Comment by AlexMyddleware** [comment]
hello @danielavetica , @JiahuiLuData yes Myddleware will be present. When it comes to the file permissions, if everything is the same, it might be because there is something wrong with the unix system install, or, because i'm working on windows at the moment, the ls command is not representing the file permissions accurately. I will run an install on ubuntu and see if i can reproduce your bug.

**Comment by danielavetica** [comment]
@AlexMyddleware awesome! as I have limited time today to work on this, let's meet up somewhere next week, I'll try to find your logo, and lets see if we can fix this together "in real life" :) (might help if I give you my laptop and let you do the magic 😄 )

**Comment by AlexMyddleware** [comment]
I'm afraid i won't be here personally, I'll get back to you asap though.

**Comment by JiahuiLuData** [comment]
Hi Alex, thanks for testing on Ubuntu, please keep me posted!

**Comment by Myddleware** [comment]
Hi @danielavetica ,
I will be at the MoodleMoot tomorrow. It will be a pleasure meeting you. 
Our technical team in France will try to reproduce your issue and help you to install Myddleware.

**Comment by danielavetica** [comment]
> Hi @danielavetica ,
> I will be at the MoodleMoot tomorrow. It will be a pleasure meeting you.
> Our technical team in France will try to reproduce your issue and help you to install Myddleware.

nice!! i am allready there, how can i recognize you? you have company clothing?;)

**Comment by Myddleware** [comment]
Hi @danielavetica,
I could install Myddleware 3.3.4 on debian 8.11, but I don't think linux version is a problem.
Could you tell me which node version you use ? You can use the command : `node -v`
For information I use Node JS version 16.20.2

Let's talk about it tomorrow at MoodleMoot ;)

**Comment by danielavetica** [comment]
Hi,

Took a little bit, but! I checked and we were using Node 18.17.1

I'll try to find you in a bit ;) 

Daniel


> Hi @danielavetica,
> 
> I could install Myddleware 3.3.4 on debian 8.11, but I don't think linux version is a problem.
> 
> Could you tell me which node version you use ? You can use the command : `node -v`
> 
> For information I use Node JS version 16.20.2
> 
> 
> 
> Let's talk about it tomorrow at MoodleMoot ;)

**Comment by Myddleware** [comment]
Hi @danielavetica ,
Hope you travelled back safely from MoodleMoot. 
I sent an invitation to you on Linkedin. Let's connect directly to solve this issue then we will write the solution on this topic.
Have a nice day.

**Comment by Myddleware** [comment]
Hi @JiahuiLuData , 
The issue for @danielavetica was just on the URL. Please try to add "index.php" in the end of the root url : 
`https://<myddleware_url>/index.php`

As this topic is pretty long and we found the solution for Daniel, please open a new issue if this solution doesn't work for you. I will close this one. 
Have a nice weekend.

**Comment by Myddleware** [comment]
Hi @JiahuiLuData ,
You can add index.php. However, the best way to solve this is to change your virtualhost by adding this : 
```
<Directory "/<your myddleware directory>">
    AllowOverride All
    Allow from All
</Directory>
```

**Comment by pratushraj** [comment]
Hi there! Is anyone solved this issue? I'm facing the same issue. Thanks

**Comment by Myddleware** [comment]
Hi @pratushraj , could you open a new issue and give us more detail about Myddleware's version and the error you get ? Thanks.
